src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,106 @@
+#include <hyro/SignalGeneratorComponent.h>
+
+using namespace hyro;
+
+namespace signal_generator
+{
+
+std::shared_ptr<hyro::HyroLogger> SignalGeneratorComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+
+hyro::Result
+SignalGeneratorComponent::init(const hyro::ComponentConfiguration &config)
+{
+  m_amplitude = 15.0;
```

Comment:
The variables `m_amplitude`, `m_frequency`, `m_cosine` and `m_name` shall be parameters. Please, use `getParameter` to set initialize these variables.